app/main/page.tsx

```diff
@@ -54,12 +56,50 @@ function App() {
     foregroundDrawer.drawHandWave(handWave);
   };
 
-  //TODO: store the new nametag content into DB
   const updateNameTagContent: SubmitHandler<NameTagContent> = (data) => {
     setNameTagContent(data);
     foregroundDrawer.drawNameTag(data);
+
+    // Update nametag in DB
+    updateNameTagInDB(data as NameTagContent);
+  };
+
+  const fetchNametagFromDB = async () => {
+    const session = await getSession();
+
+    if (session && session.user) {
+      await fetch("/api/auth/users/fetchUserData/nameTag", { 
+        method: "POST",
+        body: JSON.stringify({
+          email: session.user.email,
+        }),
+      }).then((res) => res.json()).then((resJson) => {
+        if (resJson.success && resJson.nameTag) {
+          setNameTagContent(resJson.nameTag);
+        }
+        setNameTagIsLoaded(true);
+      });
+    }
+  }
+
+  const updateNameTagInDB = async (newNameTag: NameTagContent) => {
+    const session = await getSession();
+
+    if (session && session.user) {
+      await fetch("/api/auth/users/updateUserData/nameTag", { 
+        method: "POST",
+        body: JSON.stringify({
+          email: session.user.email,
+          nameTag: newNameTag
+        }),
+      }).then((res) => res.json());
+    }
   };
```

Comment:
Can we create a component for those DB operations? It doesn't feel right to have the exact request (GET/POST) directly at the app level.

Reply:
Status on this?

Reply:
> We could move these into the NameTagForm.tsx component, since these are all related to the nametag. I'm not sure if it makes sense to put them into their own component, since they don't make up a UI component on their own, but rather are operations related to another UI component.

Per Shaomei's suggestion: will move into lib instead.

Reply:
> > We could move these into the NameTagForm.tsx component, since these are all related to the nametag. I'm not sure if it makes sense to put them into their own component, since they don't make up a UI component on their own, but rather are operations related to another UI component.
> 
> Per Shaomei's suggestion: will move into lib instead.

Done: https://github.com/aimpowered/ZoomCompanion/pull/60/commits/983af0397d7ba9a3eb19235fbca295012120bede

Reply:
> > We could move these into the NameTagForm.tsx component, since these are all related to the nametag. I'm not sure if it makes sense to put them into their own component, since they don't make up a UI component on their own, but rather are operations related to another UI component.
> 
> Per Shaomei's suggestion: will move into lib instead.

Done: https://github.com/aimpowered/ZoomCompanion/pull/60/commits/b6e63626b39b3ff57f4959dbd59288837c92bfda